client/my-sites/marketing/buttons/buttons.jsx

```diff
@@ -89,7 +85,10 @@ class SharingButtons extends Component {
 
 	UNSAFE_componentWillReceiveProps( nextProps ) {
 		// Save request has been performed
-		if ( this.props.isSaving && ! nextProps.isSaving ) {
+		if (
+			( this.props.isSavingSettings || this.props.isSavingButtons ) &&
+			! ( nextProps.isSavingSettings || nextProps.isSavingButtons )
+		) {
 			if (
 				nextProps.isSaveSettingsSuccessful &&
 				( nextProps.isSaveButtonsSuccessful || ! this.state.buttonsPendingSave )
```

Comment:
The way how this transition logic is implemented has one caveat that's worth checking: it works correctly only when the `isSavingButtons` and `isSaveButtonsSuccessful` values transition at the same time, during one atomic state update. If the transition was multi-step, like in:
```
{ isSavingButtons: true, isSaveButtonsSuccessful: false }
```
to
```
{ isSavingButtons: false, isSaveButtonsSuccessful: false }
```
to
```
{ isSavingButtons: false, isSaveButtonsSuccessful: true }
```
then the lifecycle method won't catch it and won't show the notice and update the state.

Such a two-step transition is technically perfectly valid and it's possible that the React Query mutation hook updates `isSaving` and `isSuccess` in that way. Worth checking.

Because catching state transitions in lifecycle methods often has subtle edge cases like this, it's preferrable to use a promise:
```js
saveButtons().then( () => {
  this.props.successNotice();
  this.props.markSaved();
  this.setState( { buttonsPendingSave: null } );
} );
```